scrapers/torrent.py

```diff
@@ -1,66 +1,139 @@
+            res["title"] = decode_bytes(torrent_data[b"metadata"][b"title"])
+        if b"cover url" in torrent_data[b"metadata"]:
+            res["image"] = decode_bytes(torrent_data[b"metadata"][b"cover url"])
+        if b"description" in torrent_data[b"metadata"]:
+            res["details"] = process_description_bbcode(decode_bytes(torrent_data[b"metadata"][b"description"]))
+        if b"taglist" in torrent_data[b"metadata"]:
+            res["tags"] = [{"name": decode_bytes(t)} for t in torrent_data[b"metadata"][b"taglist"]]
+        if b"taglist" in torrent_data[b"metadata"]:
+            res["performers"]=[{"name":x} for x in process_tags_performers(torrent_data[b"metadata"][b"taglist"])]
+        if b"comment" in torrent_data:
+            res["url"] = decode_bytes(torrent_data[b"comment"])
+        if b"creation date" in torrent_data:
+            res["date"] = datetime.fromtimestamp(torrent_data[b"creation date"]).strftime("%Y-%m-%d")
+    return res
+
+
+def decode_bytes(s, encodings=("utf-8", "latin-1")):
+    for enc in encodings:
+        try:
+            return s.decode(enc)
+        except UnicodeDecodeError:
+            pass
+    return s.decode("utf-8", "ignore")
+
+
+def scene_in_torrent(scene_data, torrent_data):
+    for scene in scene_data["files"]:
+        if b"length" in torrent_data[b"info"]:
+            if scene["filename"] in decode_bytes(torrent_data[b"info"][b"name"]) and torrent_data[b"info"][b"length"] == scene["size"]:
+                return True
+        elif b"files" in torrent_data[b"info"]:
+            for file in torrent_data[b"info"][b"files"]:
+                if scene["filename"] in decode_bytes(file[b"path"][-1]) and file[b"length"] == scene["size"]:
+                    return True
+
+
+def process_torrents(scene_data):
+    if scene_data:
+        for name in TORRENTS_PATH.glob("*.torrent"):
+            with open(name, "rb") as f:
+                torrent_data = bdecode(f.read())
+                if scene_in_torrent(scene_data, torrent_data):
+                    return get_torrent_metadata(torrent_data)
+    return {}
+
+def similarity_file_name(search, fileName):
+    result = difflib.SequenceMatcher(a=search.lower(), b=fileName.lower())
+    return result.ratio()
+
+def cleanup_name(name):
+    ret = str(name)
+    ret = ret.removeprefix("torrents\\").removesuffix(".torrent")
+    return ret
+
+
 if sys.argv[1] == "query":
     fragment = json.loads(sys.stdin.read())
-    title=fragment['title']
-    if '.torrent' in title:
-        lookup_torrent(title)
-    else:
-        query(title)
-    print(json.dumps(fragment))
-# Last Updated February 18, 2021
+    print(json.dumps(process_torrents(get_scene_data(fragment))))
+elif sys.argv[1] == "fragment":
+    filename = json.loads(sys.stdin.read()).get('url')
+    with open(filename, 'rb') as f:
+        torrent_data = bdecode(f.read())
+        print(json.dumps(get_torrent_metadata(torrent_data)))
```

Comment:
Is the url supposed to hold the actual filename?

Reply:
Required because after a user selects a scene in the `sceneByName` search, the filename is passed in the url variable.

I think I did some testing back then to use another field, but only url actually worked, but it's been a while ^^